src/transformStateWithClones.js

```diff
@@ -7,7 +8,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    currentState = { ...currentState };
+
+    switch (action.type) {
+      case 'addProperties': {
+        const { extraData } = action;
+
+        Object.assign(currentState, extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        const { keysToRemove } = action;
+
+        for (const key of keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        for (const key in currentState) {
+          delete currentState[key];
+        }
+        break;
+      }
+
+      default: {
+        return 'throw new Error';
```

Comment:
```suggestion
        throw new Error(`Unknown action type: ${action.type}`);
```
You should throw an error instead of returning a string